### PR TITLE
chore: readme interpolation

### LINKED ISSUE #401: Update docs for standalone language server
With the language server and coc-solidity we need to rework some of our documentation:

* [ ] simplify the repo `readme.md` to point at each of the relevant subprojects: vscode extension, coc, language server
* [ ] add a `features.md` file that lists an accurate feature set (minus foundry for now)
* [ ] modify each packaging npm task to interpolate the `features.md` information into each relevant readme
* [ ] reorganise the standalone language server readme to have features after install
* [ ] update the vscode extension readme to include a small section indicating there are options for vim/nvim/ standalone language server

### PR DESCRIPTION
**Pull request by antico5**
This PR introduces a script for interpolating markdown files, and restructuring existing readmes to extract some common content (features list). 

The way the script works is by taking `.template.md` files, interpolating the contents of the imported markdown files, and writing `.md` files. Both files (template and final) are version controlled, so when modifying a certain readme we should be modifying the template files always, since the other one is generated.

The invocation of this script is performed as a `prepackage` script on the root package, since that's the entry point for packaging and distributing our modules.

Closes #401

### CODE REVIEW

File: features.md [hunk 1]
```diff
@@ -0,0 +1,151 @@
+## Features
+
+### Code Completions
+
+**Solidity by Nomic Foundation** autocompletes references to existing symbols (e.g. contract instances, globally available variables and built-in types like arrays) and import directives (i.e. it autocompletes the path to the imported file).
```
Comment (kanej) [inline features.md:5]: This text works for the vscode extension but is out of place in the language server/coc-solidity extension.

I am not sure on this. Maybe we go for something generic `Autocomplete references` or maybe we parameterise the codegen to swap the name in (e.g. `Solidity by Nomic Foundation` or `@ignored/coc-solidity` or `Solidity language server`)

File: features.md [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+
+#### Specifying data location for variables
+
+Some types require you to specify a data location (memory, storage, calldata), depending on where they are defined. The available code actions allow the user to add, change or remove data locations depending on the error being raised.
+
+![Data location quickfix](https://raw.githubusercontent.com/NomicFoundation/hardhat-vscode/main/client/docs/gifs/data-location.gif "Specify data location")
+
+#### Fix addresses checksum
+
+The solidity compiler requires explicit addresses to be in the correct checksummed format. This quickfix transforms any address to the correct checksummed version.
```
Comment (kanej) [inline features.md:131]:
```suggestion
The solidity compiler requires explicit addresses to be in the correct checksummed format. This quickfix transforms any address to the expected format.
```

File: package.json [hunk 3]
```diff
@@ -31,6 +31,7 @@
     "prettier": "prettier *.md *.json \"{docs,syntaxes,.github,.vscode}/**/*.{md,yml,json}\" \"test/**/*.{ts,json}\"",
     "eslint": "eslint ./test/**/*.ts",
     "clean": "rimraf ./dist ./out && npm -w client run clean && npm -w server run clean",
+    "prepackage": "ts-node scripts/interpolateReadmes.ts",
```
Comment (kanej) [inline package.json:34]: This makes sense for the vscode extension release, but won't be run before the language server npm package release or the coc-solidity npm package release.

Maybe we add a `generate-readmes` task, make `prepackage` call it (safe but not necessary) and add a line to the `publish-extension.md` to invoke `generate-readmes` before bundling.

Or maybe we just need a to add `publish:coc-solidity` style commands at root and hang the readme generation off that?

**Review by kanej — COMMENTED** [review]
Can I add to the scope here and ask that the `./client/docs` get moved back to `./docs` and we move `features.md` into there (the image links will need updated).

**Comment by antico5** [comment]
@kanej ready for re-review!

**Review by kanej — APPROVED** [review]
LGTM!

**Pull request merged**